src/nti/site/site.py

```diff
@@ -50,13 +57,17 @@ def find_site_components(site_names, check_alternate=False):
     for site_name in site_names:
         if not site_name:  # Empty/default. We want the global. This should only ever be at the end
             return None
+
         if check_alternate:
             site_name = get_alternate_site_name(site_name)
             if site_name is None:
                 continue
+
         components = component.queryUtility(IComponents, name=site_name)
         if components is not None:
             return components
+    return None
```

Comment:
Is there a benefit to these explicit `return Nones`, or just good practice?

Reply:
Just best practice along the lines of "explicit is better than implicit." Plus recent versions of pylint emit a warning when some branches have a return and some don't (fall off the end). The generated bytecode is identical so there's no runtime cost.

```pycon
In [1]: import dis

In [2]: def explicit():
   ...:     return None
   ...:

In [3]: def implicit():
   ...:     pass
   ...:

In [4]: dis.dis explicit
------> dis.dis(explicit)
  2           0 LOAD_CONST               0 (None)
              3 RETURN_VALUE

In [5]: dis.dis implicit
------> dis.dis(implicit)
  2           0 LOAD_CONST               0 (None)
              3 RETURN_VALUE
```